Fix 2215 by properly remapping ports to fit new firmware changes

knandersen opened the pull request:
### Resolves

Resolves the part of #2215 that relates to block behavior, NOT connection issues.

### Proposed Changes

This PR changes port mapping as such:

Port `A`: `55` is now `0`
Port `B`: `56` is now `1`
Port `C`: `1` is now `2`
Port `D`: `2` is now `3`

Additionally it adds a `case` in `onMessage()` that catches responses from the LED to avoid warnings from appearing in the browser console. 

### Reason for Changes

LEGO has released a new firmware version 2.0.00.0017 for the LEGO BOOST Move Hub. The release included changes to the IDs that physical ports map to, and adds responses to LED commands.

knandersen committed:
Changed port-mapping to reflect firmware update.

knandersen committed:
Add case to onMessage to catch responses from the LED

knandersen commented:
Blocked by https://github.com/LLK/scratch-link/pull/120

apple502j commented:
This breaks old-firmware Boosts...

thisandagain commented:
/cc @ericrosenbaum @cwillisf @BryceLTaylor 

We should discuss how to handle communication and support for Boost hubs that are running the older firmware as @apple502j points out above.

knandersen committed:
Add port mappings to support older firmware version 2.0.00.0016 and later as well as newer firmware version 2.0.00.0017 or newer

knandersen committed:
This commit addresses a discussion in #2229 around backward compatibility.

This change creates two different port mappings for the Move Hub. By default the extension will use the newer mapping introduced in firmware version 1.0.00.0224. When connected to the hub, the extension will request the hub's firmware version. If it's older than 1.0.00.0224 it will use the older mapping.

knandersen committed:
Adjusted portmapping variables to match firmware versions correctly

thisandagain commented:
🎉 Thanks @knandersen!

ericrosenbaum commented:
Looks like ports C and D are now set incorrectly for the old firmware

knandersen committed:
Corrected old port mapping! Thanks for catching that @ericr.

ericrosenbaum commented:
Motor position reporter for port A is not working- I think it's because the `portID` of zero for port A is false-y at this line (gotta check for not null I guess!): https://github.com/LLK/scratch-vm/blob/f6f97ef3a7b562374ab9d52108c717827fe19cde/src/extensions/scratch3_boost/index.js#L1855

knandersen committed:
Fixed issue where port A motor position was not being reported because the port number was false-y

ericrosenbaum reviewed (approved):
LGTM! Tested on mac chrome with Scratch Link 1.1.46, with both old and new BOOST firmware versions.
